department-of-veterans-affairs/caseflow
# Piedram/appeals 13372

Resolves [APPEALS-13367](https://vajira.max.gov/browse/APPEALS-13367), [APPEALS-13365](https://vajira.max.gov/browse/APPEALS-13365), [APPEALS-13370](https://vajira.max.gov/browse/APPEALS-13370),
[APPEALS-13372](https://vajira.max.gov/browse/APPEALS-13372)

### Description

As a developer I need to create the action's functionality for the "Send Final Notification Letter" Task so that Contested Claim letter notifications can be tracked within Caseflow.

### Acceptance Criteria:


##### 1- Modal appears requiring user to indicate whether contesting claimant requested a different type of review option than the one selected by the appellant. Available options: Yes/No.

- [ ] If **Yes** is selected, user is required to provide additional context for this action - filed type free text: 350 character limit.
> Once confirmed:
> [ ] A docket switch task is created and active
> [ ] Task appears in the currently active tasks section and contains the following information
>  **1 Assigned to:** Clerk of the Board organization
> **2 Assigned by:** _user who completed task action_
> **3 Task Instructions:** input from the "Mark Task as Complete" modal free text field

> [ ] 4 Send Final Notification Letter” task is marked as Complete
> [ ] Task moves from Currently Active Tasks section to Case Timeline and contains the following information
> **1 Date of occurance**
> **2 Assigned to:** _user or organization_
> **3 Assigned by:** _user who completed action to assign task to COB_
> **4 Completed by:** _user who completed Send Final Notificaiton Letter task_
> [ ] All other tasks remain unchanged


[ ] If **No**, user is not required to input additional context.

> Once confirmed:
> [ ] Send Final Notification Letter” task is marked as Complete
> [ ] Task moves from Currently Active Tasks section to Case Timeline and contains the following information
> **1 Date of occurance**
> **2 Assigned to:** _user or organization_
> **Assigned by:** _user who completed action to assign task to COB_
> **Completed by:** _user who completed Send Final Notificaiton Letter task_
> [ ] All other tasks remain unchanged

### Testing Plan 1

1. Go to Caseflow login and sign in as COB member.
2. Navigate to Queue and click on the dropdown in the right corner to switch views and select **Clerk of the Board team cases**
3. Moved to `Unassigned tab`, you can see all the appeals that are on hold
4. Select any of the appeals with `Send Final Notification Letter Task` when the details page opened select in the action dropdown the option `Mark Task complete`
5. A modal windows have to show up with to option _Yes/ No_

![c1](https://user-images.githubusercontent.com/110848569/224127446-f080f976-e44c-43ab-b01b-cc92561a19ba.PNG)

6. Please select **Yes**, and fill the field with your description.

![c3](https://user-images.githubusercontent.com/110848569/224127521-eda51a2d-0289-4b64-80c8-d3ccf092a371.PNG)

7. Click on `Mark as complete`
8. The appeal appears now in the `Assigned Tab` 

![c4](https://user-images.githubusercontent.com/110848569/224128039-87ed9a91-3864-486c-93fa-f49a26dde88e.PNG)

9. Now you can see a new `Docket Switch` Task for this appeal, and the `Send Final Notification Letter Task` is completed now.

![c5](https://user-images.githubusercontent.com/110848569/224128499-bf23999d-7313-4b6a-9cfb-84ce832df994.PNG)

![c6](https://user-images.githubusercontent.com/110848569/224128549-4097d40c-51f1-4f73-9427-72101af97569.PNG)

10. If select **No** and click on `Mark as complete`.

![c2](https://user-images.githubusercontent.com/110848569/224129427-edf3b2b1-906e-40f8-8a7a-cfd7cc8f282c.PNG)

11. The appeals go to the regular caseflow any new task was created,  and `Send Final Notification Letter Task` was marked as completed.

![c8](https://user-images.githubusercontent.com/110848569/224130171-7fa89771-6cc0-4075-8bec-a49759dbe270.PNG)
